### PR TITLE
feat: use estimatedTotalAssets instead of token balance

### LINKED ISSUE #30: Test should use strategy.estimatedTotalAssets() instead of token.balanceOf(new_strategy.address)
I think the testing code should be updated to facilitate the usage.
All the tests now are doing some asserts on `token.balanceOf(new_strategy.address) `

I suggest using `strategy.estimatedTotalAssets()` combined with `pytest.approx`

@fubuloubu toughts?

### PR DESCRIPTION
**Pull request by steffenix**
fixes #30

### CODE REVIEW

File: tests/test_migration.py [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
+import pytest
+
 
 def test_migration(token, vault, strategy, amount, Strategy, strategist, gov):
     # Deposit to the vault and harvest
     token.approve(vault.address, amount, {"from": gov})
     vault.deposit(amount, {"from": gov})
     strategy.harvest()
-    assert token.balanceOf(strategy.address) == amount
+    assert pytest.approx(strategy.estimatedTotalAssets(), rel=1e-5) == amount
```
Comment (fubuloubu) [inline tests/test_migration.py:13]: can you make these constants imported from the `conftest.py` (or available as a session fixture maybe)

File: tests/test_operation.py [hunk 2]
```diff
@@ -56,17 +58,19 @@ def test_change_debt(gov, token, vault, strategy, strategist, amount):
     vault.deposit(amount, {"from": gov})
     vault.updateStrategyDebtRatio(strategy.address, 5_000, {"from": gov})
     strategy.harvest()
+    half = int(amount / 2)
 
-    assert token.balanceOf(strategy.address) == amount / 2
+    assert pytest.approx(strategy.estimatedTotalAssets(), rel=1e-5) == half
```
Comment (fubuloubu) [inline tests/test_operation.py:63]: should the `rel` be half too?

**Review by fubuloubu — APPROVED** [review]
The way I see this change is that it makes it easier when testing real strategies not to have errant failures due to slight pricing differences, and avoids needing to account for investments made during operation. Correct?

File: tests/test_operation.py (hunk 2 above)
Comment (steffenix) [reply tests/test_operation.py:63]: You have to way to use this `rel` or `abs` rel meaning relative it will "scale" with the number.

Comment (fubuloubu) [reply tests/test_operation.py:63]: ah okay

**Comment by steffenix** [comment]
> The way I see this change is that it makes it easier when testing real strategies not to have errant failures due to slight pricing differences, and avoids needing to account for investments made during operation. Correct?

First using `estimatedTotalAssets` is the way to do it as strategies invest else where they don't have the funds.
Using the estimate is the way to avoid errant failures yes.

File: tests/test_migration.py (hunk 1 above)
Comment (steffenix) [reply tests/test_migration.py]: done

File: tests/conftest.py [hunk 3]
```diff
@@ -78,3 +78,8 @@ def strategy(strategist, keeper, vault, Strategy, gov):
     strategy.setKeeper(keeper)
     vault.addStrategy(strategy, 10_000, 0, 2 ** 256 - 1, 1_000, {"from": gov})
     yield strategy
+
+
+@pytest.fixture
+def relative_aprox():
```
Comment (fubuloubu) [inline tests/conftest.py:84]: Just make it capital so it's identified as a constant. Also, might as well session scope it.
```suggestion
@pytest.fixture(scope="session")
def RELATIVE_APPROX():
```

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
If one of @poolpitako or @emilianobonassi could :+1: this change as making it easier for testing, that'd be great

File: tests/conftest.py (hunk 3 above)
Comment (steffenix) [reply tests/conftest.py]: ok, updated.

**Comment by steffenix** [comment]
ping @poolpitako @emilianobonassi

**Comment by steffenix** [comment]
Can we merge this one @fubuloubu ?

**Comment by fubuloubu** [comment]
Was hoping to get someone to take a look at this, but I see now harm in merging

**Pull request merged**